docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -0,0 +1,468 @@
+
+```jsx
+// Revalidate after 1 hour
+fetch(`https://...`, { next: { revalidate: 3600 } })
+```
+
+See the [`fetch` API reference](/docs/app/api-reference/functions/fetch) for more options.
+
+### `fetch.options.next.tag` and `revalidateTag`
+
+Next.js has a cache tagging system for fine-grained data caching and revalidation.
+
+1. When using `fetch` and `unstable_cache`, you have the option to tag cache entries with one or more tags.
+2. Then, you can call `revalidateTag` to revalidate all entries associated with that tag.
+
+For example, you can set a tag when fetching data:
+
+```jsx
+// Cache data with a tag
+fetch(`https://...`, { next: { tags: ['a', 'b', 'c'] } })
+```
+
+Then, then call `revalidateTag` with a tag to revalidate:
+
+```jsx
+// Revalidate entries with a specific tag
+revalidateTag('a')
+```
+
+There are two places you can use `revalidateTag`, depending on what you're trying to achieve:
+
+1. [Route Handlers](/docs/app/building-your-application/routing/router-handlers) - to revalidate data in response of a third party event (e.g. webhook).
+2. [Server Actions](/docs/app/building-your-application/data-fetching/server-actions) - to revalidate data after a user action (e.g. form submission).
+
+{/_ TO DO: Talk about how revalidateTag affects other caches and re-rendering.
+If you're calling it from a route handler, then it makes sense if it's for next UI visit, but for server actions, the user would expect a visual update / rerender _/}
+
+### `revalidatePath`
+
+`revalidatePath` allows you manually revalidate data **and** re-render the route segments below a specific path in a single operation. Calling `revalidatePath` methods revalidate the Data Cache, which in turn invalidates the Full Route Cache. It does not immediately invalidate the Router Cache due to the [automatic invalidation period](#duration-and-invalidation).
+
+```jsx
+revalidatePath('/')
+```
+
+There are two places you can use `revalidatePath`, depending on what you're trying to achieve:
+
+1. [Route Handlers](/docs/app/building-your-application/routing/router-handlers) - to revalidate data in response of a third party event (e.g. webhook).
+2. [Server Actions](/docs/app/building-your-application/data-fetching/server-actions) - to revalidate data after a user action (e.g. form submission).
+
+> ** `revalidatePath` vs. `router.refresh`**:
+>
+> Calling `router.refresh` will clear the Router cache, and re-render route segments on the server without invalidating the Data Cache or the Full Route Cache.
+>
+> The difference is that `revalidatePath` **revalidates** cached data in Static Routes, whereas `router.refresh` **reuses** cached data.
+
+See the [`revalidatePath` API reference]() for more information.
+
+> **Differences between `res.revalidate` vs. `revalidateTag` / `revalidatePath`**:
```

Comment:
I think this should not be explained as part of this document as it's a specific Pages Router API that only exists in `pages/api` and might feel complicated. I.e. "when do I use `res.revalidate` then?". Would be good to move this type of explanation to the upgrade guide for `pages` -> `app`

Reply:
Agreed, it was added because I've seen some issues where people were using it to try to invalidate the router cache, this was to clarify the res.revalidate is for API routes in `pages`. Thinking about it, we don't want to mention it, to not encourage people to follow any wrong patterns.